dask/array/core.py

```diff
@@ -3065,6 +3065,8 @@ def auto_chunks(chunks, shape, limit, dtype, previous_chunks=None):
         return tuple(chunks)
 
     else:
+        # Check if dtype.itemsize is greater than 0
+        assert dtype.itemsize > 0, "dtype.itemsize must be > 0"
```

Comment:
Thanks for the fix, this should be a `ValueError` instead though, as assertions don't provide the best user facing error messages (and may not run at all depending on the flags python is run with). Something like:

```suggestion
        if dtype.itemsize == 0:
            raise ValueError(
                "auto-chunking with dtype.itemsize == 0 is not supported, please pass in `chunks` explicitly"
            )       
```

Reply:
done!